staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -107,6 +109,43 @@ func (mounter *Mounter) hasSystemd() bool {
 	return *mounter.withSystemd
 }
 
+// Do a bind mount including the needed remount for applying the bind opts.
+// If the remount fails figure out if the source filesystem has the ro, nodev,
+// noexec, nosuid, noatime, relatime or nodiratime flag set and try another
+// remount with the found flags.
+func (mounter *Mounter) BindMountSensitive(mounterPath string, mountCmd string, source string, target string, fstype string, bindOpts []string, bindRemountOpts []string, bindRemountOptsSensitive []string, mountFlags []string, systemdMountRequired bool) error {
+	err := mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if err != nil {
+		return err
+	}
+	err = mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindRemountOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if err == nil {
+		return nil
+	}
+	// Check if the source has ro, nodev, noexec, nosuid, noatime, relatime,
+	// nodiratime flag...
+	var s unix.Statfs_t
+	if err := unix.Statfs(source, &s); err != nil {
+		return &os.PathError{Op: "statfs", Path: source, Err: err}
+	}
+	// ... and retry the mount with flags found above.
+	flagMapping := map[int]string {
+		unix.MS_RDONLY:     "ro",
+		unix.MS_NODEV:      "nodev",
+		unix.MS_NOEXEC:     "noexec",
+		unix.MS_NOSUID:     "nosuid",
+		unix.MS_NOATIME:    "noatime",
+		unix.MS_RELATIME:   "relatime",
+		unix.MS_NODIRATIME: "nodiratime",
+	}
+	for k, v := range flagMapping {
+		if s.Flags & k == k {
```

Comment:
Can you make sure all of this passes some sanity checks such as govet and compiles as it should?

Reply:
Looks like I found the issues. Test cases now pass.